VultisigApp/VultisigApp/Services/Sui/CetusAggregatorService.swift

```diff
@@ -0,0 +1,370 @@
+            publishedAt: "0x5cb7499fc49c2642310e24a4ecffdbee00133f97e80e2b45bca90c64d55de880",
+            version: 9
+        )
+    }
+    
+    struct CetusPackage {
+        let name: String
+        let id: String
+        let publishedAt: String
+        let version: Int
+    }
+    
+    private let baseURL: String
+    private let jsonDecoder = JSONDecoder()
+    
+    init(baseURL: String = Endpoint.cetusApiBase) {
+        self.baseURL = baseURL
+    }
+    
+    /// Find routes for swapping tokens using Cetus aggregator
+    /// - Parameters:
+    ///   - fromToken: Source token address (use "0x2::sui::SUI" for native SUI)
+    ///   - toToken: Destination token address (use USDC address for price calculation)
+    ///   - amount: Amount to swap (in smallest units)
+    /// - Returns: CetusRouteResponse containing swap routes and price information
+    /// - Throws: NSError if HTTP request fails or API returns an error
+    func findRoutes(fromToken: String, toToken: String, amount: String) async throws -> CetusRouteResponse {
+        let url = URL(string: "\(baseURL)/router_v2/find_routes")!
+        
+        // Convert amount string to UInt64 for API
+        guard let amountValue = UInt64(amount) else {
+            throw NSError(domain: "CetusAggregatorService", code: 1, userInfo: [NSLocalizedDescriptionKey: "Invalid amount format"])
+        }
+        
+        let requestBody: [String: Any] = [
+            "from": fromToken,
+            "target": toToken,  // Changed from "to" to "target"
+            "amount": amountValue,
+            "by_amount_in": true
+        ]
+        
+        var request = URLRequest(url: url)
+        request.httpMethod = "POST"
+        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.httpBody = try JSONSerialization.data(withJSONObject: requestBody)
+        request.timeoutInterval = 10
+        
+        let (data, response) = try await URLSession.shared.data(for: request)
+        
+        // Validate HTTP response status
+        guard let httpResponse = response as? HTTPURLResponse,
+              (200...299).contains(httpResponse.statusCode) else {
+            throw NSError(domain: "CetusAggregatorService", 
+                          code: (response as? HTTPURLResponse)?.statusCode ?? -1,
+                          userInfo: [NSLocalizedDescriptionKey: "HTTP request failed with status code: \((response as? HTTPURLResponse)?.statusCode ?? -1)"])
+        }
+        
+        let apiResponse = try jsonDecoder.decode(CetusAPIResponse.self, from: data)
+        
+        // Check if the API returned an error or no data
+        if apiResponse.code != 200 || apiResponse.data == nil {
+            throw NSError(domain: "CetusAggregatorService", code: apiResponse.code, userInfo: [NSLocalizedDescriptionKey: apiResponse.msg])
+        }
+        
+        return apiResponse.data!
```

Comment:
if you use 
```
guard let data = apiResponse.data else {
            throw NSError(domain: "CetusAggregatorService", code: 2, userInfo: [NSLocalizedDescriptionKey: "No data returned from API"])
        }
```
then you can just return `data` , and doesn't need to force unwrap apiResponse.data